zipkin-common/src/main/scala/com/twitter/zipkin/adjuster/MergeById.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2012 Twitter Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package com.twitter.zipkin.adjuster
+
+import com.twitter.zipkin.common._
+import scala.collection.mutable
+
+/**
+ * Merge all the spans with the same id. This is used by span stores who
+ * store partial spans and need them collated at query time.
+ *
+ * Spans without a timestamp are filtered out, as they are not possible to
```

Comment:
I think the dependencies job itself just reads all data from `traces` table. There's no index in C* implementation to only query for traces within a given timeframe. But anyway, dep job does its own merging now, so ok.